src/toplevel.c

```diff
@@ -684,53 +722,36 @@ jl_value_t *jl_toplevel_eval_flex(jl_module_t *JL_NONNULL m, jl_value_t *e, int
             }
         }
         ptls->world_age = last_age;
+        JL_GC_POP();
         return jl_nothing;
     }
-    else if (ex->head == export_sym) {
+    else if (head == export_sym) {
         for (size_t i = 0; i < jl_array_len(ex->args); i++) {
             jl_sym_t *name = (jl_sym_t*)jl_array_ptr_ref(ex->args, i);
             if (!jl_is_symbol(name))
                 jl_error("syntax: malformed \"export\" statement");
             jl_module_export(m, name);
         }
+        JL_GC_POP();
         return jl_nothing;
     }
-    else if (ex->head == global_sym) {
+    else if (head == global_sym) {
         // create uninitialized mutable binding for "global x" decl
         size_t i, l = jl_array_len(ex->args);
         for (i = 0; i < l; i++) {
-            jl_value_t *a = jl_exprarg(ex, i);
-            if (!jl_is_symbol(a) && !jl_is_globalref(a))
-                break;
-        }
-        if (i == l) {
-            for (i = 0; i < l; i++) {
-                jl_sym_t *gs = (jl_sym_t*)jl_exprarg(ex, i);
-                jl_module_t *gm = m;
-                if (jl_is_globalref(gs)) {
-                    gm = jl_globalref_mod(gs);
-                    gs = jl_globalref_name(gs);
-                }
-                assert(jl_is_symbol(gs));
-                jl_get_binding_wr(gm, gs, 0);
+            jl_sym_t *gs = (jl_sym_t*)jl_exprarg(ex, i);
```

Comment:
While you're at it, I don't think we should allow code like this anymore (saying something is a jl_sym_t, when we expect it might be a global ref. Ideally use a jl_value_t here and cast to symbol after the assert.